packages/dds/tree/src/simple-tree/prepareForInsertion.ts

```diff
@@ -96,20 +108,33 @@ export function prepareForInsertionContextless<TIn extends InsertableContent | u
 	schemaAndPolicy: SchemaAndPolicy,
 	hydratedData: Pick<FlexTreeHydratedContext, "checkout" | "nodeKeyManager"> | undefined,
 ): TIn extends undefined ? undefined : ExclusiveMapTree {
-	const mapTree = mapTreeFromNodeData(
-		data,
-		schema,
-		hydratedData?.nodeKeyManager,
-		schemaAndPolicy,
-	);
+	const mapTree = mapTreeFromNodeData(data, schema, hydratedData?.nodeKeyManager);
 
-	if (mapTree !== undefined && hydratedData !== undefined) {
-		prepareContentForHydration([mapTree], hydratedData.checkout.forest);
-	}
+	const contentArray = mapTree === undefined ? [] : [mapTree];
+	const fieldSchema = convertField(normalizeFieldSchema(schema));
+	validateAndPrepare(schemaAndPolicy, hydratedData, fieldSchema, contentArray);
 
 	return mapTree as TIn extends undefined ? undefined : ExclusiveMapTree;
 }
 
+/**
+ * If hydrating, do a final validation against the schema and prepare the content for hydration.
+ */
+function validateAndPrepare(
```

Comment:
The main change in this PR is refactoring to create this function, and use it above.

This logic combines the hydration that was already in this file (merging the two call sites into this one utility), and the validation which used to be in toMapTree that we only want to apply when hydrating (since that case is the one known to have default applied so it can actually be validated withut false positives).